src/core/services/OnChainMetaDataService.ts

```diff
@@ -0,0 +1,20 @@
+import { _getContractABI } from '@src/utils';
+import { OnChainMetaDataService, Web3Provider } from '@types';
+import {} from '@ethersproject/address';
+
+export class OnChainMetaDataServiceImpl implements OnChainMetaDataService {
+  private web3Provider: Web3Provider;
+  constructor({ web3Provider }: { web3Provider: Web3Provider }) {
+    this.web3Provider = web3Provider;
+  }
+
+  public async getContractABI(address: String, network: number): Promise<any | undefined> {
```

Comment:
we aren't using this right?

Reply:
for ENS names it looks like. @Seroxdesign ?

Reply:
Not sure what you're referring to.